### PR TITLE
Named timers

### LINKED ISSUE #41: Named timers
Currently timers can only be used to perform operations in a combined manner at an actor.

Would it feasible to let timers be named in some way so that they can trigger different events? I think tagging the timer struct with a generic type would make this work, then current behaviour is storing unit.

### PR DESCRIPTION
**Pull request by jeffa5**
Adds a `Timer` type to the `Actor` trait.
This is used for triggering different behaviours on timeout, e.g. timeouts of different peers as well as timeouts for different functionality.

It changes the timers storage from `Vec<bool>` to a `Vec<Timers>` where `Timers` is effectively a hash set.

Fixes #41

### CODE REVIEW

**Comment by jeffa5** [comment]
@jonnadal Wasn't sure if the `on_timeout` should get a `Self::Timer` or a `&Self::Timer` so happy to change it.

File: examples/timers.rs [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            PingerModelCfg {
+                server_count: 3,
+                network,
+            }
+            .into_model()
+            .checker()
+            .threads(num_cpus::get())
+            .serve(address);
+        }
+        // Some("spawn") => {
```
Comment (jonnadal) [inline examples/timers.rs:154]: I'll follow up w/ a commit to remove these lines.

**Comment by jonnadal** [comment]
Thank you @jeffa5! I did a benchmark and the performance difference is modest as well.

Before:

```
== paxos check 6 ==
Done. states=18972647, unique=9357525, depth=28, sec=16
Done. states=18972647, unique=9357525, depth=28, sec=16
Done. states=18972647, unique=9357525, depth=28, sec=16
== single-copy-register check 4 ==
Done. states=731789, unique=400233, depth=17, sec=6
Done. states=731789, unique=400233, depth=17, sec=6
Done. states=731789, unique=400233, depth=17, sec=6
== linearizable-register check 2 ==
Done. states=194646243, unique=41138199, depth=33, sec=87
Done. states=194646243, unique=41138199, depth=33, sec=86
Done. states=194646243, unique=41138199, depth=33, sec=87
== linearizable-register check 3 ordered ==
Done. states=842595255, unique=275791957, depth=49, sec=713
Done. states=842595255, unique=275791957, depth=49, sec=716
Done. states=842595255, unique=275791957, depth=49, sec=716
```

After:

```
== paxos check 6 ==
Done. states=18972647, unique=9357525, depth=28, sec=17
Done. states=18972647, unique=9357525, depth=28, sec=17
Done. states=18972647, unique=9357525, depth=28, sec=17
== single-copy-register check 4 ==
Done. states=731789, unique=400233, depth=17, sec=5
Done. states=731789, unique=400233, depth=17, sec=6
Done. states=731789, unique=400233, depth=17, sec=6
== linearizable-register check 2 ==
Done. states=194646243, unique=41138199, depth=33, sec=91
Done. states=194646243, unique=41138199, depth=33, sec=92
Done. states=194646243, unique=41138199, depth=33, sec=92
== linearizable-register check 3 ordered ==
Done. states=842595255, unique=275791957, depth=49, sec=744
Done. states=842595255, unique=275791957, depth=49, sec=774
Done. states=842595255, unique=275791957, depth=49, sec=767
```

> @jonnadal Wasn't sure if the `on_timeout` should get a `Self::Timer` or a `&Self::Timer` so happy to change it.

I don't feel strongly about it either. Let's keep as-is for now.

**Review by jonnadal — APPROVED** [review]
No blockers. Thanks again!

**Pull request merged**